src/ncp/ncp_base_mtd.cpp

```diff
@@ -1372,26 +1372,20 @@ template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_THREAD_PENDING_DATASE
 
 template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_THREAD_ACTIVE_DATASET_TLVS>(void)
 {
-    otError                  error = OT_ERROR_NONE;
     otOperationalDatasetTlvs dataset;
 
-    SuccessOrExit(error = otDatasetGetActiveTlvs(mInstance, &dataset));
-    SuccessOrExit(error = mEncoder.WriteData(dataset.mTlvs, dataset.mLength));
-
-exit:
-    return error;
+    dataset.mLength = 0;
+    IgnoreError(otDatasetGetActiveTlvs(mInstance, &dataset));
+    return mEncoder.WriteData(dataset.mTlvs, dataset.mLength);
```

Comment:
While this is also probably correct and works, I would suggest doing it like this:
```suggestion
    if (otDatasetGetActiveTlvs(mInstance, &dataset)) != OT_ERROR_NONE)
    { 
          dataset.mLength = 0;
    }
    return mEncoder.WriteData(dataset.mTlvs, dataset.mLength);
```

This way, we are not making any assumption about behavior of `otDatasetGetActiveTlvs()` when it returns an error (that it would not change its input `dataset` when returning an error).